### PR TITLE
sixth part 

### PR DESCRIPTION
**Pull request by Baldinini**

### CODE REVIEW

File: src/main/java/mate/academy/controllers/AddCarController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/car/creation.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String carModel = req.getParameter("car_model");
+        Long manufacturerId = Long.valueOf(req.getParameter("manufacturer_id"));
```
Comment (yurwing) [inline src/main/java/mate/academy/controllers/AddCarController.java]: it would be good to check input data(if data is empty or letters for id)

**Review by yurwing — APPROVED** [review]

File: src/main/java/mate/academy/controllers/AddCarController.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/mate/academy/controllers/AddCarController.java]: @YuraBondarenko this is checked on JSPs

File: src/main/java/mate/academy/controllers/GetAllDriversController.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+public class GetAllDriversController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate.academy");
+    private final DriverService driverService =
+            (DriverService) injector.getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Driver> allDrivers = driverService.getAll();
+
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controllers/GetAllDriversController.java]: redundant empty line

File: src/main/java/mate/academy/controllers/GetAllManufacturersController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+public class GetAllManufacturersController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate.academy");
+    private final ManufacturerService manufacturerService =
+            (ManufacturerService) injector.getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Manufacturer> allManufacturer = manufacturerService.getAll();
+
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controllers/GetAllManufacturersController.java]: same

File: src/main/webapp/web.xml [hunk 4]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allDrivers</servlet-name>
+        <servlet-class>mate.academy.controllers.GetAllDriversController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allDrivers</servlet-name>
+        <url-pattern>/driver/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/</url-pattern>
```

File: src/main/webapp/web.xml [hunk 5]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>driversCreation</servlet-name>
+        <servlet-class>mate.academy.controllers.AddDriverController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>driversCreation</servlet-name>
+        <url-pattern>/driver/creation</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 6]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteDriver</servlet-name>
+        <servlet-class>mate.academy.controllers.DeleteDriverController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteDriver</servlet-name>
+        <url-pattern>/driver/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/delete</url-pattern>
```
same comments for car and manufacturer controllers

File: src/main/webapp/web.xml [hunk 7]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+    
+    <servlet>
+        <servlet-name>addDriverToCar</servlet-name>
+        <servlet-class>mate.academy.controllers.AddDriverToCarController</servlet-class>
+    </servlet>
+    
+    <servlet-mapping>
+        <servlet-name>addDriverToCar</servlet-name>
+        <url-pattern>/car/driver/add</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/drivers/add</url-pattern>
```
is it a good practice to use plurals in URLs

File: src/main/webapp/WEB-INF/views/car/all.jsp [hunk 8]
```diff
@@ -0,0 +1,63 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/car/all.jsp]: remove this empty line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
overall good job!

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**